File: frontend/src/types/lan/LanDiscoveryStateMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

/**
 * Tells the server whether this device takes part in LAN discovery. With
 * `enabled` false the server sends no peer list and leaves the device out of
 * the lists of the other devices in the same network.
 */
export class LanDiscoveryStateMessage implements ITypedMessage {
    public readonly type = MessageType.LAN_DISCOVERY_STATE;
    public msg: {
        enabled: boolean;
    };

    public constructor(msg: { enabled: boolean }) {
        this.msg = msg;
    }
}

File: backend/backend/src/LanComponent/Common/Api/DTOs/LanDiscoveryStateMessage.cs
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.Api.DTOs;

namespace backend.LanComponent.Common.Api.DTOs;

/// <summary>
/// Announces whether the sending client takes part in LAN discovery.
/// A client with discovery switched off receives no peer list and appears in no other client's list.
/// A connection starts out switched off, so a client is discoverable from its own report onwards.
/// </summary>
public class LanDiscoveryStateMessage : ITypedMessage
{
    public static string TypeString => "lan-discovery-state";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;

    [JsonPropertyName("enabled")]
    public required bool Enabled { get; set; }
}
